torch/_subclasses/fake_tensor.py

```diff
@@ -56,36 +58,49 @@ def _is_tensor_constructor(func: OpOverload):
         len(schema.returns) == 1 and schema.returns[0].type is torch._C.TensorType.get()
     )
 
-
 # Similar to `MetaConverter`, this is a class for converting
 # multiple tensors into fake tensors which share the same view/storage
 # structure. Like `MetaConverter`, it will keep alive all
 # tensors that are converted to FakeTensors.
-class FakeTensorConverter(MetaConverter):
+class FakeTensorConverter(object):
+    tensor_memo: weakref.WeakValueDictionary
+    meta_converter: MetaConverter
+
     def __init__(self):
-        self.tensor_memo = {}
+        # FakeTensors store the FakeTensorMode which in turn stores a
+        # FakeTensor, so we need to hold a weak reference to the FakeTensor
+        # otherwise we would induce a circular reference
+        self.tensor_memo = weakref.WeakValueDictionary()
         self.meta_converter = MetaConverter()
 
-    def from_real_tensor(self, t):
+    def _get_memo(self, t):
+        if t in self.tensor_memo:
+            out = self.tensor_memo[t]
+            out._fix_weakref()
+            return out
+        return None
+
+    def from_real_tensor(self, fake_mode, t):
+        if self._get_memo(t) is not None:
+            return self.tensor_memo[t]
```

Comment:
This looks wrong shouldn't you return the results of get memo

Reply:
wheres the walrus operator where u need it (doesnt work python 3.7)